src/application/components/GitHubReleaseHoverCard.tsx

```diff
@@ -0,0 +1,175 @@
+import { useEffect, useReducer, type ReactNode } from "react";
+import { HoverCard } from "./HoverCard";
+import { Badge } from "./Badge";
+import { Markdown } from "./Markdown";
+import { Spinner } from "./Spinner";
+import IconOutlink from "@apollo/icons/small/IconOutlink.svg";
+import IconGitHub from "@apollo/icons/default/IconGitHubSolid.svg";
+
+interface GitHubReleaseHoverCardProps {
+  children?: ReactNode;
+  version: string;
+}
+
+export function GitHubReleaseHoverCard({
+  children,
+  version,
+}: GitHubReleaseHoverCardProps) {
+  if (version.startsWith("0.0.0")) {
+    return children;
+  }
+
+  return (
+    <HoverCard openDelay={0}>
+      <HoverCard.Trigger asChild>{children}</HoverCard.Trigger>
+      <HoverCard.Content>
+        <CardContents version={version} />
+      </HoverCard.Content>
+    </HoverCard>
+  );
+}
+
+type ReleaseState =
+  | { status: "pending"; data: null }
+  | { status: "success"; data: ReleaseData }
+  | { status: "error"; data: null };
+
+type Action =
+  | { type: "load" }
+  | { type: "success"; payload: ReleaseData }
+  | { type: "failed" };
+
+function reducer(state: ReleaseState, action: Action): ReleaseState {
+  switch (action.type) {
+    case "load":
+      return { status: "pending", data: null };
+    case "success":
+      return { status: "success", data: action.payload };
+    case "failed":
+      return { status: "error", data: null };
+    default:
+      return state;
+  }
+}
+
+function CardContents({ version }: { version: string }) {
+  const [{ status, data }, dispatch] = useReducer(reducer, {
+    status: "pending",
+    data: null,
+  });
+
+  useEffect(() => {
+    let ignored = false;
+    dispatch({ type: "load" });
+    getGitHubRelease(version).then(
+      (data) => {
+        if (ignored) {
+          return;
+        }
+
+        dispatch({ type: "success", payload: data });
+      },
+      () => {
+        if (!ignored) {
+          dispatch({ type: "failed" });
+        }
+      }
+    );
+
+    return () => {
+      ignored = true;
+    };
+  }, [version]);
+
+  if (status === "pending") {
+    return (
+      <div className="flex items-center justify-center min-w-80 min-h-80">
+        <Spinner size="lg" />
+      </div>
+    );
+  }
+
+  if (status === "error") {
+    return "Error";
+  }
+
+  const { release, latest } = data;
+
+  return (
+    <div className="flex flex-col gap-8">
+      <header className="flex flex-col gap-2 bg-primary dark:bg-primary-dark">
+        <h2 className="text-2xl text-heading dark:text-heading-dark font-heading font-medium flex items-center gap-2">
+          <IconGitHub className="size-6" />
+          {release.name}{" "}
+          {release.prerelease ? (
+            <Badge variant="beta">Pre-release</Badge>
+          ) : latest ? (
+            <Badge variant="success">Latest</Badge>
+          ) : (
+            <Badge variant="warning">Outdated</Badge>
```

Comment:
The publish date would also be very interesting here, but I believe our GH releases don't always match up, so this would have to come from npm?

Reply:
> but I believe our GH releases don't always match up

The only thing I can think of was the one-off scenario when we created GH releases for 3.8 prerelease versions after the fact, but in general GH and npm publish dates should match

Reply:
<img width="410" alt="Screenshot 2024-05-16 at 11 33 01 AM" src="https://github.com/apollographql/apollo-client-devtools/assets/565661/43c1f832-f065-4ea9-910d-fd85b8190216">

Done 🙂

Reply:
Amazing! :)